mobile/lib/main.dart

```diff
@@ -212,20 +213,31 @@ class ImmichAppState extends ConsumerState<ImmichApp>
     final router = ref.watch(appRouterProvider);
     final immichTheme = ref.watch(immichThemeProvider);
 
-    return MaterialApp(
-      localizationsDelegates: context.localizationDelegates,
-      supportedLocales: context.supportedLocales,
-      locale: context.locale,
-      debugShowCheckedModeBanner: true,
-      home: MaterialApp.router(
-        title: 'Immich',
-        debugShowCheckedModeBanner: false,
-        themeMode: ref.watch(immichThemeModeProvider),
-        darkTheme: getThemeData(colorScheme: immichTheme.dark),
-        theme: getThemeData(colorScheme: immichTheme.light),
-        routeInformationParser: router.defaultRouteParser(),
-        routerDelegate: router.delegate(
-          navigatorObservers: () => [TabNavigationObserver(ref: ref)],
+    return ProviderScope(
+      overrides: [
+        localeProvider.overrideWithValue(context.locale),

```

Comment:
What is the purpose of this `overrides` property?

Reply:
It sets the `localeProvider` value for the rest subtree of widgets every time that `context.locale` gets updated. Do you have an alternative way of settings the initial value of `localeProvider` and then updating it?